Framework/Algorithms/src/Rebin.cpp

```diff
@@ -355,5 +355,10 @@ void Rebin::propagateMasks(API::MatrixWorkspace_const_sptr inputWS,
   }
 }
 
+Parallel::ExecutionMode Rebin::getParallelExecutionMode(
+    const std::map<std::string, Parallel::StorageMode> &storageModes) const {
+  return Parallel::getCorrespondingExecutionMode(storageModes.begin()->second);
```

Comment:
This is what I had previously done in a new base class `TriviallyParallelAlgorithm`. I probably still want to add a class like this, but for the time being I am postponing it until we have a cleared understanding or some of the issues we will be facing.